lib/driver.rb

```diff
@@ -0,0 +1,44 @@
+require 'csv'
+
+module RideShare
+    class Driver
+        DRIVER_INFO = CSV.read('support/drivers.csv')
+
+        attr_accessor :driver_id, :name, :vin
+
+        def initialize(id)
+            raise ArgumentError, "#{id} is not a valid ID number, only positive whole numbers are accepted" unless id.is_a?(Integer) && id >= 1
+            raise ArgumentError, "#{id} Cannot be found" unless (DRIVER_INFO.map { |line| line[0].to_i }).include? id
+
+            DRIVER_INFO.each do |line|
+                next unless line[0].to_i == id
+                @driver_id = id
+                @name = line[1]
+                @vin = line[2]
+            end
+
+            raise ArgumentError, "#{vin} is an invalid VIN number" unless @vin.length == 17
+        end
+
+        def trips
+            Trip.driver_trips(@driver_id)
+        end
+
+        def avg_rating
+            @driver_trips = trips
```

Comment:
I'm curious why you chose to use an instance variable here. Seems like the trips don't need to persist long term, only for the duration of the method, which to me indicates that a local variable might be a better choice.